internal/server/questions.go

```diff
@@ -123,6 +122,17 @@ const htmlHead = `<!DOCTYPE html><html lang="en"><head>
             }
 
             console.error("error ", response.status, response.statusText);
+						
+						let textMessage = "Something went wrong. Please try again or report back to us.";
+						const errorResponse = await response.json();
+						let showButton = true;
+
+						if(errorResponse && errorResponse.message) {
+							if(errorResponse.message === "stepRunEnded") {
+								textMessage = "Sorry this HIT already ended, you cannot submit it to MTurk.";
+								showButton = false;
+							}
+						}
```

Comment:
I'm not super clear what's going on here.

The rule is, we should submit to MTurk even if the hit did timeout, because we want them to change screen and show the error MTurk message. Is this what's happening? I don't want them to get blocked on our iframe.

Also, what's going on with indentation? 😆

Reply:
Oh my apologize, i miss interpreted our last discussion, i will fix this one. Will fix the indentation as well 😅